confmap/marshaler.go

```diff
@@ -0,0 +1,24 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package confmap
+
+// Marshaler defines an optional interface for custom configuration marshalling.
+// A configuration struct can implement this interface to override the default
+// marshalling.
+type Marshaler interface {
+	// Marshal the config into a Conf in a custom way.
+	// The Conf will be empty and can be merged into.
+	Marshal(component *Conf) error
+}
```

Comment:
We can have this in the same file as unmarshaler.

Reply:
It's been moved into the same confmap file.